WcaOnRails/app/controllers/api/v0/competitions_controller.rb

```diff
@@ -22,6 +22,21 @@ def show
     render json: competition
   end
 
+  def results
+    competition = competition_from_params
+    render json: competition.results
+  end
+
+  def competitors
+    competition = competition_from_params
+    render json: competition.competitors
+  end
+
+  def registrations
+    competition = competition_from_params
+    render json: competition.registrations
```

Comment:
Given that's public we should only show accepted registrations (`competition.registrations.accepted`). Also some of the attributes are confidential (e.g. `ip`, `comments`) and should be stripped out.
We should probably include events for each registration, so I'd suggest defining a special `serializable_hash` method for the `Registration` model, similarly to the [competition one](https://github.com/thewca/worldcubeassociation.org/blob/805d6ddbff5d55bd8ec67f8477efc579b8212de5/WcaOnRails/app/models/competition.rb#L1343-L1360), so that the returned hash includes only the public/relevant attributes and the corresponding list of event ids. Something along the lines of:
```rb
  def serializable_hash(options = nil)
    {
      id: id,
      competition_id: competition_id,
      user_id: user_id,
      event_ids: events.map(&:id)
    }
  end
```
And then we should load all the `events` with a single query like that:
```rb
render json: competition.registrations.accepted.includes(:events)
```